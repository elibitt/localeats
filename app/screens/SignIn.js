import React from "react";
import { View } from "react-native";
import { Card, Button, Input } from "react-native-elements";
import { onSignIn } from "../auth";

export default ({ navigation }) => (
	<View style={{ paddingVertical: 20 }}>
		<Card>
			<Input label="Email" placeholder="Email address..." />
			<Input label="Password" secureTextEntry placeholder="Password..." />
			<Button
				buttonStyle={{ marginTop: 20, backgroundColor: "#03A9F4" }}
				title="SIGN IN"
				onPress={() => {
					onSignIn().then(() => navigation.navigate("SignedIn"));
				}}
			/>
		</Card>
	</View>
);