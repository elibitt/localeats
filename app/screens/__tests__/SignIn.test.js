import React from 'react';
import renderer from 'react-test-renderer';

import SignIn from '../SignIn';

test('renders correctly', () => {
  const tree = renderer.create(<SignIn />).toJSON();
  expect(tree).toMatchSnapshot();
});
