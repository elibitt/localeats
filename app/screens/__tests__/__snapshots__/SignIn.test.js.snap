// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 1334,
        "left": 0,
        "paddingTop": 120,
        "top": 0,
        "width": 750,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
          undefined,
        ]
      }
    />
    <View
      accessible={false}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "bottom": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(255, 255, 255, .95)",
                "borderRadius": 10,
                "paddingBottom": 22,
                "paddingTop": 32,
                "width": 720,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                  "marginBottom": 30,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "black",
                      "fontSize": 30,
                    }
                  }
                >
                  Welcome back!
                </Text>
              </View>
              <View
                style={
                  Object {
                    "marginLeft": 10,
                    "marginTop": -10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "black",
                      "fontSize": 20,
                      "marginTop": 10,
                    }
                  }
                >
                  Sign in to continue...
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderBottomColor": "rgba(0, 0, 0, 0.38)",
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 1,
                    "borderColor": "#86939e",
                    "flexDirection": "row",
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 40,
                      "justifyContent": "center",
                      "marginLeft": 15,
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                      theme={
                        Object {
                          "colors": Object {
                            "disabled": "hsl(208, 8%, 90%)",
                            "divider": "#bcbbc1",
                            "error": "#ff190c",
                            "grey0": "#393e42",
                            "grey1": "#43484d",
                            "grey2": "#5e6977",
                            "grey3": "#86939e",
                            "grey4": "#bdc6cf",
                            "grey5": "#e1e8ee",
                            "greyOutline": "#bbb",
                            "platform": Object {
                              "android": Object {
                                "error": "#f44336",
                                "primary": "#2196f3",
                                "secondary": "#9C27B0",
                                "success": "#4caf50",
                                "warning": "#ffeb3b",
                              },
                              "ios": Object {
                                "error": "#ff3b30",
                                "primary": "#007aff",
                                "secondary": "#5856d6",
                                "success": "#4cd964",
                                "warning": "#ffcc00",
                              },
                            },
                            "primary": "#2089dc",
                            "searchBg": "#303337",
                            "secondary": "#8F0CE8",
                            "success": "#52c41a",
                            "warning": "#faad14",
                          },
                        }
                      }
                      underlayColor="white"
                    >
                      <Text />
                    </View>
                  </View>
                </View>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  autoFocus={false}
                  keyboardAppearance="light"
                  keyboardType="email-address"
                  onChangeText={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="Email"
                  returnKeyType="next"
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "black",
                      "flex": 1,
                      "fontSize": 18,
                      "marginLeft": 10,
                      "minHeight": 40,
                    }
                  }
                  testID="RNE__Input__text-input"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "borderBottomColor": "rgba(0, 0, 0, 0.38)",
                  "marginTop": 16,
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 1,
                    "borderColor": "#86939e",
                    "flexDirection": "row",
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 40,
                      "justifyContent": "center",
                      "marginLeft": 15,
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                      theme={
                        Object {
                          "colors": Object {
                            "disabled": "hsl(208, 8%, 90%)",
                            "divider": "#bcbbc1",
                            "error": "#ff190c",
                            "grey0": "#393e42",
                            "grey1": "#43484d",
                            "grey2": "#5e6977",
                            "grey3": "#86939e",
                            "grey4": "#bdc6cf",
                            "grey5": "#e1e8ee",
                            "greyOutline": "#bbb",
                            "platform": Object {
                              "android": Object {
                                "error": "#f44336",
                                "primary": "#2196f3",
                                "secondary": "#9C27B0",
                                "success": "#4caf50",
                                "warning": "#ffeb3b",
                              },
                              "ios": Object {
                                "error": "#ff3b30",
                                "primary": "#007aff",
                                "secondary": "#5856d6",
                                "success": "#4cd964",
                                "warning": "#ffcc00",
                              },
                            },
                            "primary": "#2089dc",
                            "searchBg": "#303337",
                            "secondary": "#8F0CE8",
                            "success": "#52c41a",
                            "warning": "#faad14",
                          },
                        }
                      }
                      underlayColor="white"
                    >
                      <Text />
                    </View>
                  </View>
                </View>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  blurOnSubmit={true}
                  keyboardAppearance="light"
                  onChangeText={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="Password"
                  returnKeyType="done"
                  secureTextEntry={true}
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "black",
                      "flex": 1,
                      "fontSize": 18,
                      "marginLeft": 10,
                      "minHeight": 40,
                    }
                  }
                  testID="RNE__Input__text-input"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "borderRadius": 10,
                  "flex": 0,
                  "marginTop": 32,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(232, 147, 142, 1)",
                      "borderColor": "#2089dc",
                      "borderRadius": 10,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "height": 50,
                      "justifyContent": "center",
                      "padding": 8,
                      "width": 200,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "paddingBottom": 1,
                        "paddingTop": 2,
                        "textAlign": "center",
                      }
                    }
                  >
                    SIGN IN
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
